src/ceph-volume/ceph_volume/util/device.py

```diff
@@ -30,8 +35,21 @@ class Devices(object):
     def __init__(self, devices=None):
         if not sys_info.devices:
             sys_info.devices = disk.get_devices()
-        self.devices = [Device(k) for k in
-                            sys_info.devices.keys()]
+        self.devices = []
+        with futures.ThreadPoolExecutor() as executor:
```

Comment:
have you considered using max_workers. The default is quite high and in my tests leads to longer runtimes. When I ran with max_workers=4 I was getting consistently shorter runtimes (.5->1s less) - same result as when I ran the threads directly in my PR.

Reply:
Yep, can confirm that. 
|                     | max_workers=0 | max_workers=2 | max_workers=4 | max_workers=8 |
|------------------|-----------------------|-----------------------|-----------------------|-----------------------|
|avg runtime in ms over 10 runs |2366.1|2396.3|2189.7|2362.7|